### PR TITLE
Refactor: documentation part 1

### PR DESCRIPTION
**Pull request by ninaandrup**
Related to #60

In this PR
- [x] Introduction
- [x] Functions
- [x] Data Types
- [x] Lists
- [x] Records
- [x] References
- [x] Arrays
- [x] Namespaces
- [x] Concurrency
- [x] Effects
- [x] Fixpoints
- [x] Interoperability
- [x] Build & Packages
- [x] Tips and Tricks

There are still open issues for some of the pages. But everything not in those issues should be fixed in this PR.

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Good work so far 👍

**Comment by ninaandrup** [comment]
I'm a bit unsure of this example:

```
def map(f: a -> b & e, l: List[a]): List[b] & e = ...
```
The code in the flix library is `& ef` instead of `& e`

**Comment by magnus-madsen** [comment]
We have decided to use `ef` (instead of `e`) by convention.

**Comment by ninaandrup** [comment]
[Here ](https://github.com/flix/programming-flix/blob/6864d39b90922b78f8d3e5be7ae26b915e367166/src/content/Namespaces.js#L249) Eq is not defines for the enum. Does it matter, since it is illustrating `Using Enums from a Namespace` ?

**Comment by magnus-madsen** [comment]
> [Here ](https://github.com/flix/programming-flix/blob/6864d39b90922b78f8d3e5be7ae26b915e367166/src/content/Namespaces.js#L249) Eq is not defines for the enum. Does it matter, since it is illustrating `Using Enums from a Namespace` ?

You can sneak in a `with Eq` on the relevant enum.

**Comment by ninaandrup** [comment]
I think maybe [this paragraph](https://github.com/flix/programming-flix/blob/6864d39b90922b78f8d3e5be7ae26b915e367166/src/content/Effects.js#L72) needs to be rewritten. I could not find any functions in the library using an argument `f: a ~> Unit`. Everything is changed to `f: a -> Unit & ef`, right?

**Comment by ninaandrup** [comment]
> I think maybe [this paragraph](https://github.com/flix/programming-flix/blob/6864d39b90922b78f8d3e5be7ae26b915e367166/src/content/Effects.js#L72) needs to be rewritten. I could not find any functions in the library using an argument `f: a ~> Unit`. Everything is changed to `f: a -> Unit & ef`, right?

Trying it out in `VS code`, I can see, that `f: a ~> Unit` won't compile.
On the same page, there is already [this paragraph](https://github.com/flix/programming-flix/blob/6864d39b90922b78f8d3e5be7ae26b915e367166/src/content/Effects.js#L100) about `effect polymorphism`. Maybe the other paragraph should just inform the user, that specifying an argument function as `impure` is not possible?

**Comment by ninaandrup** [comment]
> > I think maybe [this paragraph](https://github.com/flix/programming-flix/blob/6864d39b90922b78f8d3e5be7ae26b915e367166/src/content/Effects.js#L72) needs to be rewritten. I could not find any functions in the library using an argument `f: a ~> Unit`. Everything is changed to `f: a -> Unit & ef`, right?
> 
> Trying it out in `VS code`, I can see, that `f: a ~> Unit` won't compile. On the same page, there is already [this paragraph](https://github.com/flix/programming-flix/blob/6864d39b90922b78f8d3e5be7ae26b915e367166/src/content/Effects.js#L100) about `effect polymorphism`. Maybe the other paragraph should just inform the user, that specifying an argument function as `impure` is not possible?

Mentioned again [here in the table](https://github.com/flix/programming-flix/blob/6864d39b90922b78f8d3e5be7ae26b915e367166/src/content/Effects.js#L229).

**Comment by ninaandrup** [comment]
Should I open some new issues for the paragraphs needing to be rewritten and such?

**Comment by magnus-madsen** [comment]
> Should I open some new issues for the paragraphs needing to be rewritten and such?

Yes. Lets keep this PR for simple things and then make a new issue for the things where new sentences are needed.

**Comment by magnus-madsen** [comment]
Can you open a new ticket for the fixpoint issue? Then I will ask Jonathan to take a look.

**Comment by ninaandrup** [comment]
> Can you open a new ticket for the fixpoint issue? Then I will ask Jonathan to take a look.


Sure :) see #67

File: src/content/Introduction.js [hunk 1]
```diff
@@ -41,21 +41,21 @@ class Introduction extends React.Component {
                 <CodeBlock>
                     {`/// An algebraic data type for shapes.
 enum Shape {
-    case Circle(Int),        // circle radius
-    case Square(Int),        // side length
-    case Rectangle(Int, Int) // height and width
+    case Circle(Int32),        // circle radius
+    case Square(Int32),        // side length
```
Comment (magnus-madsen) [inline src/content/Introduction.js:45]: Fix indentation such that these line up again.

File: src/content/TipsAndTricks.js [hunk 2]
```diff
@@ -550,13 +550,13 @@ Some(x + y) `}</CodeBlock>
                         For example, we can cast an impure expression to a pure expression:
                     </p>
 
-                    <CodeBlock>{`Console.println("Hello World") as Unit & Pure`}</CodeBlock>
+                    <CodeBlock>{`Console.printLine("Hello World") as Unit & Pure`}</CodeBlock>
```
Comment (magnus-madsen) [inline src/content/TipsAndTricks.js:553]: I think we should just `println` no need for the full `Console.printLine ... `

File: src/content/TipsAndTricks.js [hunk 3]
```diff
@@ -550,13 +550,13 @@ Some(x + y) `}</CodeBlock>
...
 
                     <p>
                         As a short-hand, we can simply write:
                     </p>
 
-                    <CodeBlock>{`Console.println("Hello World") as & Pure`}</CodeBlock>
+                    <CodeBlock>{`Console.printLine("Hello World") as & Pure`}</CodeBlock>
```
Comment (magnus-madsen) [inline src/content/TipsAndTricks.js:559]: same comment here

**Comment by magnus-madsen** [comment]
- Very happy to have this!
- Minor comments can be fixed in subsequent PR.

**Pull request merged**